File: python3/p_003.py
import util


"""
Find the largest prime factor of a given number, n
n = 600851475143
"""


class p_3(util.Problem):
    def __init__(self):
        super().__init__()
        self.problem_id = "3"

    def solve(self):
        n = 600851475143
        return self.f(n)

    def f(self, n):
        i = 2
        while i * i <= n:
            if n % i:
                i += 1
            else:
                n //= i
        return n


if __name__ == "__main__":
    instance = p_3()
    instance.run()

File: java/src/main/p_003.java
package main;

/*
What is the largest prime factor of the number
600851475143
 */

public class p_003 implements util.IProblem {

    public static void main(String[] args) {
        util.time(new p_003());
    }

    public long run () {
        long n = 600851475143L;
        int max = -1;

        for (int i = 2; i < (int) Math.sqrt(n)+1; i++ ) {
            if (n % i == 0) {
                if (i > max && util.isPrime(i)) max = i;
            }
        }

        return max;
    }
}
